openshift-kube-apiserver/admission/customresourcevalidation/imagedigestmirrorset/validate_imagedigestmirrorset.go

```diff
@@ -0,0 +1,114 @@
+package imagedigestmirrorset
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	v1 "github.com/openshift/api/config/v1"
+	operatorsv1alpha1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1alpha1"
+	"github.com/openshift/library-go/pkg/apiserver/admission/admissionrestconfig"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const (
+	PluginName   = "config.openshift.io/ValidateImageDigestMirrorSet"
+	IcspKind     = "ImageContentSourcePolicy"
+	IcspResource = "imagecontentsourcepolicies"
+	IdmsKind     = "ImageDigestMirrorSet"
+	IdmsResource = "imagedigestmirrorsets"
+	ItmsKind     = "ImageTagMirrorSet"
+	ItmsResource = "imagetagmirrorsets"
```

Comment:
Nit: I'd probably keep ICSP, IDMS and ITMS since all of those are shortcuts and based on naming scheme it'll be appropriate to user uppercase for all of those. Also, you're already using upercase (example `validateICSPUse` above).